core/controllers/base.py

```diff
@@ -282,6 +282,39 @@ def render_downloadable_file(self, values, filename, content_type):
             'attachment; filename=%s' % filename)
         self.response.write(values)
 
+    def render_static_template(self, filepath, iframe_restriction='DENY'):
+        """Prepares a Jinja-less HTML response to be sent to the client.
+           Args:
+            filepath: str. The template filepath.
+            iframe_restriction: str or None. Possible values are
+                'DENY' and 'SAMEORIGIN':
+
+                DENY: Strictly prevents the template to load in an iframe.
+                SAMEORIGIN: The template can only be displayed in a frame
+                    on the same origin as the page itself.
+        """
+
+        self.response.cache_control.no_cache = True
+        self.response.cache_control.must_revalidate = True
+        self.response.headers['Strict-Transport-Security'] = (
+            'max-age=31536000; includeSubDomains')
+        self.response.headers['X-Content-Type-Options'] = 'nosniff'
+        self.response.headers['X-Xss-Protection'] = '1; mode=block'
+
+        if iframe_restriction is not None:
+            if iframe_restriction in ['SAMEORIGIN', 'DENY']:
+                self.response.headers['X-Frame-Options'] = iframe_restriction
+            else:
+                raise Exception(
+                    'Invalid X-Frame-Options: %s' % iframe_restriction)
+
+        self.response.expires = 'Mon, 01 Jan 1990 00:00:00 GMT'
+        self.response.pragma = 'no-cache'
```

Comment:
From this line up the code is the same as in `render_template()`. Make a new method named `prepare_headers()` that will handle this.

Reply:
Done